tests/storage/clone.go

```diff
@@ -120,6 +175,35 @@ var _ = SIGDescribe("[Serial]VirtualMachineClone Tests", func() {
 		return out
 	}
 
+	createSnapshotAndWaitForReady := func(vmName, vmNamespace string) *v1alpha1.VirtualMachineSnapshot {
+		var err error
+
+		snapshot := &v1alpha1.VirtualMachineSnapshot{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "snapshot-" + vmName,
+				Namespace: vmNamespace,
+			},
+			Spec: v1alpha1.VirtualMachineSnapshotSpec{
+				Source: k8sv1.TypedLocalObjectReference{
+					APIGroup: pointer.String("kubevirt.io"),
+					Kind:     "VirtualMachine",
+					Name:     vmName,
+				},
+			},
+		}
+
+		snapshot, err = virtClient.VirtualMachineSnapshot(snapshot.Namespace).Create(context.Background(), snapshot, v1.CreateOptions{})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+		EventuallyWithOffset(1, func() bool {
```

Comment:
same here with waitSnapshotReady
I'll mention that 1) there is probably a wait snapshot ready in the snapshot tests.. 2) There is an initiative to make error more clear for example this PR: https://github.com/kubevirt/kubevirt/pull/8509/commits/8210de28bdef64a5caedea3c9510c198a1f4636e
It makes the error show the full status in the error and not only something like `expected false to be true`

Reply:
So:
1) You're right. The reason I didn't use snapshot's functions is that they rely on global variables which makes them unusable outside of snapshot tests' scope. I think that we should try to never should rely on global variables for this reason. Anyhow - I think that it would be best to refactor snapshot's functions, but I think it should be done in a follow-up PR.

2) I fully agree that errors should be described in a human-readable way. The PR you referenced seems to do it perfectly, and thanks for introducing me to `Satisfy()` - pretty cool! In this specific case, I've added a description to the `Eventually()` clause: `"snapshot should be ready"` (which didn't exist in snapshot's function). Therefore, I think that in this specific case using `Satisfy()` won't make it more readable / help to reuse code since I already have a description there and this ready condition is only used once in the code.